.github/workflows/Tests.yaml

```diff
@@ -515,3 +548,142 @@ jobs:
+            echo "Tag 'v$RELEASE_VERSION' already exists"
+            exit 1
+          fi
+          if [ $(git rev-parse HEAD) == $(git rev-parse origin/release) ]; then
+            echo "Nothing changed since last release $(git describe release)."
+            exit 1
+          fi
+        shell: bash
+      - name: Reserve Zenodo DOI and prepare repository
+        run: |
+          python3 .github/scripts/Release.py prepare -vv \
+            --version $RELEASE_VERSION \
+            --zenodo-token ${{ secrets.ZENODO_READWRITE_TOKEN }} \
+            --zenodo-sandbox \
+            --github-token ${{ secrets.GITHUB_TOKEN }}
+          git diff
+      # Push a commit with the new version to the branch we're on (should be
+      # `develop` unless the workflow was triggered manually on another branch).
+      # The push won't trigger the workflow again because GitHub prevents
+      # actions from triggering workflows (this uses the default GITHUB_TOKEN).
+      - name: Commit and push
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git commit -a -m "Prepare for release $RELEASE_VERSION"
+          git show HEAD
+          git status
+          git push
+          git push origin HEAD:release
+      - name: Compile release notes
+        run: |
+          python3 tools/CompileReleaseNotes.py -v -o release_notes.md \
+            --github-token ${{ secrets.GITHUB_TOKEN }}
+      - name: Tag and release
+        uses: actions/create-release@v1
+        env:
+          # If we need this event to trigger other workflows we could use a
+          # personal access token
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: v${{ env.RELEASE_VERSION }}
+          release_name: Release ${{ env.RELEASE_VERSION }}
+          body_path: release_notes.md
+      # The release is now public on GitHub. We build the release documentation
+      # from the archive and also build an executable to make sure the archive
+      # is functional.
+      - name: Build release documentation and an executable
+        working-directory: /work
+        run: |
+          wget https://github.com/${{ github.repository }}/archive/v${RELEASE_VERSION}.tar.gz -O spectre.tar.gz
+          tar -xzf spectre.tar.gz && mv spectre-* spectre
+          mkdir build && cd build
+          cmake \
+            -D CMAKE_C_COMPILER=clang \
+            -D CMAKE_CXX_COMPILER=clang++ \
+            -D CMAKE_Fortran_COMPILER=gfortran-8 \
+            -D CHARM_ROOT=/work/charm_6_10_2/multicore-linux-x86_64-clang \
+            -D OVERRIDE_ARCH=x86-64 \
+            -D CMAKE_BUILD_TYPE=Release \
+            -D DEBUG_SYMBOLS=OFF \
+            ../spectre
+          make doc-check
+          make -j2 ExportCoordinates1D
+          ctest -R InputFiles.ExportCoordinates.Input1D.yaml.execute
+      - name: Deploy to gh-pages
```

Comment:
Question: should we try to keep documentation for old versions? So that, e.g.
```
spectre-code.org/v2020.11.01
```
would give you the documentation for that? I worry that with monthly releases we will quickly have too much data. The dox on my machine is currently 52MB...

Reply:
Yea that would certainly be nice.. not sure if it's really necessary though when we advise people to use the latest release. I've also been thinking about printing the release notes to the docs..

Reply:
:+1:
Keeping a "Version History" or "Release Notes" would be nice. I don't think that needs to be (or should be done) as part of this PR. I think it's better to get this merged sooner and then see what releases look like and are doing, then adjust what we don't like :)